crates/network/src/ethereum/builder.rs

```diff
@@ -146,6 +120,37 @@ impl TransactionBuilder<Ethereum> for alloy_rpc_types::TransactionRequest {
     }
 }
 
+/// Build an unsigned transaction
+pub(crate) fn build_unsigned<N>(request: TransactionRequest) -> BuilderResult<N::UnsignedTx>
+where
+    N: Network,
+    N::UnsignedTx: From<TxLegacy> + From<TxEip1559> + From<TxEip2930> + From<TxEip4844Variant>,
+{
+    match (
+        request.gas_price.as_ref(),
+        request.max_fee_per_gas.as_ref(),
+        request.access_list.as_ref(),
+        request.max_fee_per_blob_gas.as_ref(),
+        request.blob_versioned_hashes.as_ref(),
+        request.sidecar.as_ref(),
+    ) {
+        // Legacy transaction
+        (Some(_), None, None, None, None, None) => build_legacy(request).map(Into::into),
```

Comment:
while we're touching this code, we should decompose this into a set of functions `fn will_build_legacy() -> bool` that capture our criteria for building each style of transaction. then this becomes a fallthrough if

I also think our criteria here are not most-friendly to users.

- if a sidecar is set, we should ALWAYS build 4844, regardless of if legacy gas is indicated
- if `max_fee_per_blob_gas` is set, but no sidecar/blob is set, that should be a build error
- 1559 should be our default tx type in the fallthrough case. legacy should be chosen only if gas_price is set.


function should look something as follows:
```
if self.will_build_4844() { self.build_4844() } 
else if self.will_build_2930() { self.build_2930() }
else if self.will_build_legacy() { self.build_legacy() }
else { self.build_1559() } 
```

Reply:
opened as #423

Reply:
is this PR blocked by #423?

Reply:
no, should be separate work

Reply:
Is something missing for this PR to be merged?